src/model/results-check.ts

```diff
@@ -22,9 +22,18 @@ const ResultsCheck = {
       files.map(async filepath => {
         if (!filepath.endsWith('.xml')) return;
         core.info(`Processing file ${filepath}...`);
-        const fileData = await ResultsParser.parseResults(path.join(artifactsPath, filepath));
-        core.info(fileData.summary);
-        runs.push(fileData);
+        try {
+          const content = fs.readFileSync(path.join(artifactsPath, filepath), 'utf8');
+          if (!content.includes('<test-run')) {
+            // noinspection ExceptionCaughtLocallyJS
+            throw new Error('File does not appear to be a NUnit XML file');
+          }
+          const fileData = await ResultsParser.parseResults(path.join(artifactsPath, filepath));
+          core.info(fileData.summary);
+          runs.push(fileData);
+        } catch (error: any) {
+          core.warning(`Failed to parse ${filepath}: ${error.message}`);
+        }
       }),
     );
```

Comment:
Looks good to me.

Just a thought, feel free to ignore if you think it's not worth it:

Reading from NUnit docs, `<test-run>` is the required root element for any NUnit 3.0 test result file. Not sure if it's worth the optimisation, as I don't know how big these files get, but we could consider doing a test that just matches the start of the file

```ts
function isValidTestRunFile(xmlContent) {
    // Match optional XML declaration, followed by <test-run as the root element
    return /^\s*(<\?xml\b[^>]*\?>)?\s*<test-run\b/.test(xmlContent)
}
```

Reply:
> as I don't know how big these files get

They can get arbitrarily large, depending how many tests are ran. My project runs 25k+ tests, resulting in 30mb+ file size.

Reply:
Maybe we can go with something like this:

```typescript
try {
  const filePath = path.join(artifactsPath, filepath);
  const fd = fs.openSync(filePath, 'r');
  const bufferSize = 4096; // Read the first 4KB
  const buffer = Buffer.alloc(bufferSize);
  const bytesRead = fs.readSync(fd, buffer, 0, bufferSize, 0);
  fs.closeSync(fd);

  const contentStart = buffer.toString('utf8', 0, bytesRead);
  if (!contentStart.includes('<test-run')) {
    throw new Error('File does not appear to be a NUnit XML file');
  }

  const fileData = await ResultsParser.parseResults(filePath);
  core.info(fileData.summary);
  runs.push(fileData);
} catch (error: any) {
  core.warning(`Failed to parse ${filepath}: ${error.message}`);
}
```

I did not test this, but the idea would be to read the first 4KB of the file to look for `<test-run` so we don't end up reading 30mb+ files in the CI. Might be early optimization, but if we already know this can happen, I think this would be an acceptable solution. I don't think we need to go too deep in this rabbit hole for now as long as it's better than before.

@coderabbitai what do you think? :P

Reply:
Since this is already merged, I moved this to its own issue: https://github.com/game-ci/unity-test-runner/issues/288 Feel free to create a PR for it :+1: